docs/jwst/assign_wcs/main.rst

```diff
@@ -6,59 +6,69 @@ Description
 :Alias: assign_wcs
 
 
-``jwst.assign_wcs`` is run in the beginning of the level 2B JWST pipeline.
+The ``assign_wcs`` step is run at the beginning of the Stage 2 pipelines - :ref:`calwebb_image2 <calwebb_image2>`
+and :ref:`calwebb_spec2 <calwebb_spec2>` - for both imaging and spectroscopic exposures.
 It associates a WCS object with each science exposure. The WCS object transforms
 positions in the detector frame to positions in a world coordinate frame - ICRS and wavelength.
 In general there may be intermediate coordinate frames depending on the instrument.
 The WCS is saved in the ASDF extension of the FITS file. It can be accessed as an attribute of
-the meta object when the fits file is opened as a data model.
+the meta object when the FITS file is opened as a data model.
 
 The forward direction of the transforms is from detector to world coordinates
-and the input positions are 0-based.
+and the input pixel coordinates are 0-indexed.
 
-``jwst.assign_wcs`` expects to find the basic WCS keywords in the
-SCI header. Distortion and spectral models are stored in reference files in the
+The ``assign_wcs`` step expects to find the basic WCS keywords in the
+"SCI" extension header of the input FITS file. Distortion and spectral models are stored in reference files in the
 `ASDF <http://asdf-standard.readthedocs.org/en/latest/>`__  format.
 
 For each observing mode, determined by the value of ``EXP_TYPE`` in the science header,
-assign_wcs retrieves reference files from CRDS and creates a pipeline of transforms from
+``assign_wcs`` retrieves reference files from CRDS and creates a pipeline of transforms from
 input frame ``detector`` to a frame ``v2v3``. This part of the WCS pipeline may include
 intermediate coordinate frames. The basic WCS keywords are used to create
 the transform from frame ``v2v3`` to frame ``world``.
 
 For image display with software like DS9 that relies on specific WCS information, a SIP-based
 approximation to the WCS is fit. The results are FITS keywords stored in
 ``model.meta.wcsinfo``. This is not an exact fit, but is accurate to ~0.25 pixel and is sufficient
-for display purposes. This step, which occurs for imaging modes early, is performed by default but
-can be switched off, and parameters controlling the fit can also be adjusted.
+for display purposes. This step, which occurs for imaging modes, is performed by default, but
+can be switched off, and parameters controlling the SIP fit can also be adjusted.
 
+For each exposure, ``assign_wcs`` uses reference files and WCS header keywords
+to create the WCS object. What reference files are retrieved
+from CRDS is determined based on `EXP_TYPE` and other keywords in the science file header.
 
-``jwst.assign_wcs`` is based on `gwcs <https://gwcs.readthedocs.io/en/latest/>`__ and
-uses `asdf <http://asdf.readthedocs.io/en/latest/>`__.
+The ``assign_wcs`` step can accept the single slope image that is the result of averaging
```

Comment:
might be good to give the specific names of the datamodels that are accepted here

Reply:
I've reworded to reference ``rate`` and ``rateints`` products, and removed any mention of TSO, because ``rateints`` is acceptable for several modes, not just TSO.